Assets/Scripts/LevelSelect/FloatingNotification.cs

```diff
@@ -0,0 +1,49 @@
+using TMPro;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FloatingNotification : MonoBehaviour
+{
+    public const float fadeDuration = 0.5f;
+    public const float totalDuration = 1.5f;
+    public const float floatVelocity = 1f;
+
+    [SerializeField] private TextMeshPro textMesh;
+    [SerializeField] private Rigidbody2D rb;
+
+    void Awake()
+    {
+        rb.velocity = Vector2.up * floatVelocity;
+        textMesh = GetComponentInChildren<TextMeshPro>();
+        textMesh.alpha = 0f;
+    }
+
+    public void Initialize(Vector2 position, string text) {
+        transform.position = position;
+        GetComponentInChildren<TMPro.TMP_Text>().text = text;
+        StartCoroutine(AnimateCoroutine());
+    }
+
+    private IEnumerator AnimateCoroutine() {
+        yield return StartCoroutine(Fade(0f, 1f));
+        yield return new WaitForSeconds(totalDuration - fadeDuration * 2);
+        yield return StartCoroutine(Fade(1f, 0f));
+        Destroy(gameObject);
+    }
+
+    private IEnumerator Fade(float from, float to, float duration = fadeDuration)
+    {
+        float t = 0f;
+        while (t < duration)
+        {
+            var value = t / duration;
+            t += Time.deltaTime;
+            textMesh.alpha = Mathf.Lerp(from, to, value);
```

Comment:
nit: a different easing curve like`Mathf.SmoothStep` might look better when dealing with fading text

```suggestion
            textMesh.alpha = Mathf.SmoothStep(from, to, value);
```

Reply:
OK